azext_iot/central/providers/devicetwin_provider.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from knack.log import get_logger
+from azext_iot.common.utility import find_between
+from azext_iot._factory import _bind_sdk
+from azext_iot.common.shared import SdkType
+from azext_iot.common.utility import unpack_msrest_error
+from azext_iot.common.sas_token_auth import BasicSasTokenAuthentication
+
+
+logger = get_logger(__name__)
+
+
+class CentralDeviceTwinProvider:
+    def __init__(self, cmd, app_id: str, device_id: str):
+        """
+        Provider for devicetwin APIs
+
+        Args:
+            cmd: command passed into az
+            app_id: name of app (used for forming request URL)
+            token: (OPTIONAL) authorization token to fetch device details from IoTC.
+                MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+                Useful in scenarios where user doesn't own the app
+                therefore AAD token won't work, but a SAS token generated by owner will
+        """
+        self._cmd = cmd
+        self._app_id = app_id
+        self._device_id = device_id
+
+    def get_device_twin(self, central_dns_suffix):
+        from azext_iot.common._azure import get_iot_central_tokens
+
+        tokens = get_iot_central_tokens(self._cmd, self._app_id, central_dns_suffix)
+
+        exception = None
+
+        # The device could be in any hub associated with the given app.
+        # We must search through each IoT Hub until device is found.
+        for token_group in tokens.values():
+            sas_token = token_group["iothubTenantSasToken"]["sasToken"]
+            endpoint = find_between(sas_token, "SharedAccessSignature sr=", "&sig=")
+            target = {"entity": endpoint}
+            auth = BasicSasTokenAuthentication(sas_token=sas_token)
+            service_sdk, errors = _bind_sdk(target, SdkType.service_sdk, auth=auth)
```

Comment:
Not blocking...I know its how it worked before - just a mental note, `_bind_sdk` is planned to go away. I can take care of this later with some planned refactors.

Reply:
Sounds good, happy to assist with that whenever the work comes down the...pipeline